sdk/PrebidServerAdapter/PBServerAdapter.m

```diff
@@ -105,17 +105,19 @@ - (void)requestBidsWithAdUnits:(nullable NSArray<PBAdUnit *> *)adUnits
 
                     for (int i = 0; i< bidsArray.count; i++) {
                         NSMutableDictionary *adServerTargetingCopy = [bidsArray[i][@"ext"][@"prebid"][@"targeting"] mutableCopy];
-                        if (i == 0) {
-                            NSString *cacheId = cacheIds[i];
-                            adServerTargetingCopy[kAPNAdServerCacheIdKey] = cacheId;
+                        if (adServerTargetingCopy != nil) {
+                            if (i == 0) {
+                                NSString *cacheId = cacheIds[i];
+                                adServerTargetingCopy[kAPNAdServerCacheIdKey] = cacheId;
+                            }
+                            NSString *bidderCacheId = cacheIds[i];
+                            NSString *cacheIdkey =[ NSString stringWithFormat:@"%@_%@", kAPNAdServerCacheIdKey, bidsArray[i][@"seat"]];
+                            cacheIdkey = cacheIdkey.length > 20 ? [cacheIdkey substringToIndex:20] : cacheIdkey;
+                            adServerTargetingCopy[cacheIdkey] = bidderCacheId;
+                            PBBidResponse *bidResponse = [PBBidResponse bidResponseWithAdUnitId:adUnitId adServerTargeting:adServerTargetingCopy];
+                            PBLogDebug(@"Bid Successful with rounded bid targeting keys are %@ for adUnit id is %@", [bidResponse.customKeywords description], adUnitId);
+                            [bidResponsesArray addObject:bidResponse];
                         }
-                        NSString *bidderCacheId = cacheIds[i];
-                        NSString *cacheIdkey =[ NSString stringWithFormat:@"%@_%@", kAPNAdServerCacheIdKey, bidsArray[i][@"seat"]];
-                        cacheIdkey = cacheIdkey.length > 20 ? [cacheIdkey substringToIndex:20] : cacheIdkey;
-                        adServerTargetingCopy[cacheIdkey] = bidderCacheId;
-                        PBBidResponse *bidResponse = [PBBidResponse bidResponseWithAdUnitId:adUnitId adServerTargeting:adServerTargetingCopy];
-                        PBLogDebug(@"Bid Successful with rounded bid targeting keys are %@ for adUnit id is %@", [bidResponse.customKeywords description], adUnitId);
-                        [bidResponsesArray addObject:bidResponse];
                     }
                     [delegate didReceiveSuccessResponse:bidResponsesArray];;
```

Comment:
Why are u sending `didReceiveSuccessResponse ` when targeting keys are nil?

Reply:
check out the full code. I'm not sending `didReceiveSuccessResponse` when targeting keys are nil, that's inside bid response retrieval.